### ISSUE TITLE
Courier doesn't detect nodes picked by Macro Parameter udi pickers

### ISSUE THREAD

**Issue by bowserm** [umbraco/Umbraco.Courier.Issues#8]
I have reproduced this on an Umbraco 7.7.9 site with Courier 3.1.6. The problem happens on macros with macro parameters that pick udi values. These are the two that I have reproduced with:
- `Umbraco.ContentPicker2`
- ` Umbraco.MediaPicker2`

When one of these macros is embedded in a rich text editor and a value is picked, Courier does not pick up on the picked content as a dependency.

In my tests, I have reproduced the problem using a `Umbraco.TinyMCEv3` RTE. I have also confirmed that in my `/config/courier.config` I have the following config:
```
  <macros>
    <macrosAreDependencies>true</macrosAreDependencies>
    <macroParametersAreDependencies>true</macroParametersAreDependencies>

    <!-- Propertytypes that CAN contain macro mark-up (like the ones inserted with TinyMCE) -->
    <add key="TinyMCE3">Umbraco.TinyMCEv3</add>
    <add key="TextboxMultiple">Umbraco.TextboxMultiple</add>
    <add key="Textstring">Umbraco.Textbox</add>
    <add key="Simple Editor">Umbraco.MarkdownEditor</add>
    <add key="Macro Container">Umbraco.MacroContainer</add>
  </macros>
  <templates>
    <add key="include" />
  </templates>
</itemDataResolvers>

<macroPropertyTypeResolvers>
  <contentPickers>
    <!-- Macro Property Types, that store Content IDs, to link to media or content -->
    <add key="contentPicker">Umbraco.ContentPickerAlias</add>
    <add key="contentPicker2">Umbraco.ContentPicker2</add>
    <add key="multiplemediaPicker">Umbraco.MultipleMediaPicker</add>
    <add key="mediaPicker">Umbraco.MediaPicker</add>
    <add key="mediaPicker2">Umbraco.MediaPicker2</add>
```

I have also confirmed that when I attempt to courier the content, I see the macro itself in the `queue.xml` file as follows, so I am pretty sure that the problem is not in my `courier.config` or with the macro itself. The problem appears to be with the macro parameters that pick udi values:

```
<QueuedItemIdentifier>
    <ItemId>
        <Id>RecentNewsTeaser</Id>
        <ProviderId>2ab40e30-e292-11df-85ca-0800200c9a66</ProviderId>
    </ItemId>
    <Processed>true</Processed>
    <Skipped>true</Skipped>
    <ItemType>Dependency</ItemType>
    <Level>1</Level>
    <MaxDependencyLevel>2147483647</MaxDependencyLevel>
    <Name>Recent News Teaser</Name>
</QueuedItemIdentifier>
```

I can confirm that courier does find nodes picked by the obsolete content and media pickers that pick by node id. For now, that seems to be the workaround. For all of our clients that use Courier, we need to make sure that all of their macro parameter pickers are of the obsolete variety that pick by node id and not by udi until this can be resolved.

Thanks!

**Comment by clausjensen** [comment]
Hey @bowserm 
Thanks for reporting!
This issue will be fixed in the 3.1.8 release of Courier. I'm not sure when it will be released but I'm attaching a nightly build here, that you can try using in the meantime.

There's a couple of other fixes in there as well, so if you encounter something strange that didn't happen before, please let us know :)

[Courier.v3.1.8-alpha-20190604-001.zip](https://github.com/umbraco/Umbraco.Courier.Issues/files/3251588/Courier.v3.1.8-alpha-20190604-001.zip)
[Courier.v3.1.8-alpha-20190604-001.HotFix.zip](https://github.com/umbraco/Umbraco.Courier.Issues/files/3251590/Courier.v3.1.8-alpha-20190604-001.HotFix.zip)
[Courier.Concorde.UI.v3.1.8-alpha-20190604-001.zip](https://github.com/umbraco/Umbraco.Courier.Issues/files/3251589/Courier.Concorde.UI.v3.1.8-alpha-20190604-001.zip)
[Courier.Core.3.1.8-alpha-20190604-001.nupkg.zip](https://github.com/umbraco/Umbraco.Courier.Issues/files/3251594/Courier.Core.3.1.8-alpha-20190604-001.nupkg.zip)

If you're running Courier on a normal setup you should just be using the first or second one .. first one is the "package" version and the second one is for manually updating. Concorde is for Courier running in Cloud (hopefully not many who would need this one anymore).

**Comment by bowserm** [comment]
Any news about when this will be released? The fix seems to be working for two of our clients who have faced this problem, but I don't see the Courier 3.1.8 build in the list of nightly builds yet: http://nightly.umbraco.org/

Thanks

**Comment by bowserm** [comment]
I'm still not seeing this Courier.Core.3.1.8 nightly build in either of these feeds. It is still the only way that I've been able to fix our clients issues deploying macros that use content or media picker macro parameters. Am I looking in the right place?

- http://nightly.umbraco.org/?container=umbraco-courier
- http://nightly.umbraco.org/?container=umbraco-courier-release

**Comment by martina-umbraco** [comment]
Hello @clausjensen , I would like to know which is the current umbraco courier stable version (3.16. or 3.1.7) and when 3.1.8 will be released.

Thank you,
Martina

**Comment by clausjensen** [comment]
@bowserm @martina-umbraco 
Sorry about the delay - we have finally released the 3.1.8 build. It is available for manual updates on 
http://nightly.umbraco.org/?container=umbraco-courier-release + on Our + MyGet.

I am not sure when/if Cloud will be automatically upgrading any projects using it, as we are currently trying to actually get every site moved to Deploy instead. If you do have a project using it, you can however upgrade it manually for now if you experience issues.

**Comment by martina-umbraco** [comment]
> 
> 
> @bowserm @martina-umbraco
> Sorry about the delay - we have finally released the 3.1.8 build. It is available for manual updates on
> http://nightly.umbraco.org/?container=umbraco-courier-release + on Our + MyGet.
> 
> I am not sure when/if Cloud will be automatically upgrading any projects using it, as we are currently trying to actually get every site moved to Deploy instead. If you do have a project using it, you can however upgrade it manually for now if you experience issues.

Thank you Claus, much appreciated!
I'm using Umbraco on Azure so I'll need to upgrade it manually as for documentation.
I have a question though: I still don't see the changelog for 3.1.7 and 3.1.8 here: https://our.umbraco.com/packages/umbraco-pro/umbraco-courier/

Am I looking in the right place?

Thanks again,
Martina

**Comment by clausjensen** [comment]
@martina-umbraco Yep you are 😄  .. I've updated that page with the changes for latest two versions as well: https://our.umbraco.com/packages/umbraco-pro/umbraco-courier/
